Notify/Notify/Notify/AppShell.xaml.cs

```diff
@@ -106,82 +100,70 @@ private void setMessagingCenterStopServiceMessageSubscription()
                     {
                         Device.BeginInvokeOnMainThread(() =>
                         {
                             try
                             {
-                                arrivedDestinationForTheFirstTime();
+                                AnnounceDestinationArrival();
                             }
                             catch (Exception ex)
                             {
-                                Debug.WriteLine("Failed in MessagingCenter.Subscribe<LocationMessage>: " + ex.Message);
+                                Debug.WriteLine($"Failed in MessagingCenter.Subscribe<Location>: {ex.Message}");
                             }
                         });
                     }
+                    
+                    m_LastUpdatedLocation = location;
+                    Debug.WriteLine($"Updated last updated location: {m_LastUpdatedLocation}");
                 }
             });
         }
         
-        private async Task<double> sendCurrentLocationToAzureFunction(LocationMessage currentLocation)
+        bool isCurrentLocationFarEnoughForUpdating(Location location)
         {
-            double distance = -1;
+            bool shouldUpdate;
 
-            try
+            if (m_LastUpdatedLocation == null)
             {
-                dynamic request = new JObject();
-                request.location = new JObject();
-                request.location.latitude = currentLocation.Latitude;
-                request.location.longitude = currentLocation.Longitude;
-                string json = JsonConvert.SerializeObject(request);
-                StringContent content = new StringContent(json, Encoding.UTF8, "application/json");
-
-                // Send an HTTP POST request to the Azure Function
-                HttpResponseMessage response = await httpClient.PostAsync("distance", content).ConfigureAwait(false);
-                response.EnsureSuccessStatusCode();
-
-                // Read the response JSON
-                string responseJson = await response.Content.ReadAsStringAsync();
-                dynamic output = JsonConvert.DeserializeObject(responseJson);
-
-                // Extract the distance value from the output JSON
-                distance = Convert.ToDouble(output.distance);
+                shouldUpdate = true;
             }
-            catch (Exception ex)
+            else
             {
-                DisplayAlert("Error", $"An error occurred: {ex.Message}", "OK");
+                double distance = GeolocatorUtils.CalculateDistance(
+                    latitudeStart: location.Latitude, longitudeStart: location.Longitude,
+                    latitudeEnd: location.Latitude, longitudeEnd: m_LastUpdatedLocation.Longitude,
+                    units: GeolocatorUtils.DistanceUnits.Kilometers) * 1000;
+            
+                Debug.WriteLine($"Distance from last updated location: {distance} meters");
+                shouldUpdate = distance > 30;
```

Comment:
Consider exporting `30` to Constants 
We can call it: `DISTANCE_THRESHOLD` or something like that.